src/wwwroot/directives/link-toggler.js

```diff
@@ -0,0 +1,34 @@
+(function () {
+    'use strict';
+  
+    angular
+      .module('dopplerRelay')
+      .directive('linkToggler', linkToggler);
+  
+      linkToggler.$inject = [
+      "auth"
+    ];
+  
+    function linkToggler(auth) {
+      var directive = {
+        restrict: 'A',
+        link: function (scope, element, attrs, ctrls) {
+            var decodedTkn = auth.decodeToken();
```

Comment:
I think that you should avoid to call _decodeToken()_.

Permissons object should be already available.